pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageContainerImpl.java

```diff
@@ -163,6 +166,11 @@ private ByteBuf getCompressedBatchMetadataAndPayload() {
         int uncompressedSize = batchedMessageMetadataAndPayload.readableBytes();
         ByteBuf compressedPayload = compressor.encode(batchedMessageMetadataAndPayload);
         batchedMessageMetadataAndPayload.release();
+
+        int delta = compressedPayload.capacity() - extraCapacity;
+        extraCapacity = compressedPayload.capacity();
+        producer.client.getMemoryLimitController().forceReserveMemory(delta);
```

Comment:
It's not a good idea to reserve memory in a "get" method. We should always consider that a "get" method can be called anytime.

Reply:
OK, fixed